boomonkeyboo/scratch_project
# Added verification of JWT tokens, routing related to them, and refreshing tokens

**Issue Type**


**Description**

Registering a user now creates a JWT token. Most user actions now verify that JWT to check if it's still valid. Those routes now also refresh the JWT token, making sure the user has an updated one.
Also naturally allows testing with Postman once again.

**Steps to Validate Feature**

1. Using Postman, send a POST request to either /api/registeruser or /api/userlogin with the respective valid info. Ex. Logging in with 
```javascript
{
    "username":"tw",
    "password":"newpassword"
} 
``` 
2. Navigate to Headers, and check that the JWT cookie exists.
3. Then either:

- send a GET request to /api/getallchatrooms or /api/getallchatlogs
- send a POST request to /api/addchatroom or /api/addchatlog

(Make sure to include the proper information with every request, as well as the username!)

4. See the correct info given back to you.
5. Wait 60 seconds after your last request to send another. This should instead tell you there was a problem, as expected.

**Previous behavior**
Would only create a JWT token upon logging in.

**Expected behavior**
Verifies JWT with most client-side requests, and gives the client an updated token.
